keps/sig-docs/20191020-doc-policies-for-third-party-content.md

```diff
@@ -0,0 +1,339 @@
+  - [Design Details](#design-details)
+    - [Graduation Criteria](#graduation-criteria)
+- [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+<!-- /toc -->
+
+**Note:** This KEP does not target any release; SIG Docs follows a continuous
+release process for website content.
+
+## Summary
+
+This KEP seeks consensus on how Kubernetes docs handle two types of content:
+
+1. Content from or about third-party providers ("third-party content")
+
+Minimize and eliminate third-party content except when necessary for Kubernetes 
+to function in-project.
+
+2. Content hosted on multiple sites ("dual-sourced content")
+
+Minimize and eliminate dual-sourced content except when necessary for Kubernetes
+to function in-project.
+
+**Note:** This KEP defines "in project" to mean projects in the Kubernetes org, 
+which includes the [kubernetes](https://github.com/kubernetes) and 
+[kubernetes-sigs](https://github.com/kubernetes) repositories.
+
+## Introduction
+
+Kubernetes documentation teaches Kubernetes users about how
+Kubernetes works, how to use in-project Kubernetes features, and how to
+build on top of Kubernetes infrastucture.
+
+Feature docs are not a place for vendor pitches. Nonetheless, SIG Docs sometimes
+receives pull requests to place advertising-like content on the Kubernetes
+website. Some PRs clearly do not belong in feature docs, but other
+instances are less clear.
+
+Feature docs also contain dual-sourced content. A good practice for code
+project docs is to host single-sourced content only, and to provide
+links to other providers’ single-sourced content. This simplifies
+version management and reduces the work required to maintain content.
+
+This KEP proposes that Kubernetes documentation sets these boundaries:
+
+- Kubernetes documentation includes 
+
+This KEP defines how to handle third-party and dual-sourced content in 
+documentation, so that authors can
+judge what is appropriate to propose and so that PR approvers can make
+consistent, fair decisions during the review process.
+
+## Motivation
+
+SIG Docs publishes Kubernetes
+documentation on kubernetes.io in line with its
+[charter](https://github.com/kubernetes/community/blob/master/sig-docs/charter.md#scope)
+and sets standards for website content. Prior to this KEP, there are no
+clear guidelines or standards for third-party and dual-sourced content.
+
+The Kubernetes documentation is currently a mix of both 1) documentation
+describing the Kubernetes open source project; and 2) content describing
+how to install or use Kubernetes on several third party Kubernetes
+offerings.
+
+Some third party content is necessary in order for Kubernetes to
+function. For example: container runtimes (containerd, CRI-o, Docker), 
+networking policy (CNI plugins), Ingress controllers, and logging all require 
+third party components. Pages like [Logging Using Elasticsearch and Kibana](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/) 
```

Comment:
@thockin 
> So what is the principle we want?

We have language that talks about [linking versus hosting](https://github.com/kubernetes/enhancements/pull/1327/files#diff-7c573870e6b3f25f38e3837aaaa193dcR206-R207) in order to avoid exactly this scenario, where _no one_ knows whether a doc is fresh. However, linking is no guarantee of freshness: links break, redirect unhelpfully, and go stale too. We need to future proof and stale proof our content as best we're able. Drawing boundaries at third party content seems like the best way to do that.

Reply:
The principle here is that third party docs should only be included where absolutely necessary. The E+K doc says it is specific to GCE and is out of date from what I can tell (and therefore would be eligible for removal unless it was updated, or better linked externally). If we are only using this as an example to highlight the principle though, maybe something more black and white like [Logging Using Stackdriver](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-stackdriver/) would be better. Again, this doesn't change the principle or intent, but is a more clear example without having to read the linked document. Swapping the link would be sufficient.